\begin{appendices}

%Some Table of Contents entry formatting
\addtocontents{toc}{\protect\renewcommand{\protect\cftchappresnum}{\appendixname\space}}
\addtocontents{toc}{\protect\renewcommand{\protect\cftchapnumwidth}{6em}}

%Begin individual appendices, separated as chapters

\chapter{Abbreviations and definitions}

\section{Badapple}
\label{appendix:badapple}

\begin{table}
\caption{Abbreviations (Badapple)}
\begin{tabular}{p{0.2\linewidth}p{0.8\linewidth}}
\hline
\makecell[r]{\textbf{BA}} & Batting average\\
\makecell[r]{\textbf{BAO}} & BioAssay Ontology\\
\makecell[r]{\textbf{BARD}} & BioAssay Research Database\\
\makecell[r]{\textbf{HTS}} & High-throughput screening\\
\makecell[r]{\textbf{ML}} & Machine learning\\
\makecell[r]{\textbf{MLP}} & Molecular Libraries Program\\
\makecell[r]{\textbf{MLSMR}} & Molecular Libraries Small Molecule Repository\\
\makecell[r]{\textbf{Ro5}} & Rule of 5\\
\makecell[r]{\textbf{SAR}} & Structureâ€“activity relationships\\
\makecell[r]{\textbf{UNMCMD}} & University of New Mexico Center for Molecular Discovery\\
\hline
\end{tabular}
\end{table}

\section{KGAP}
\label{appendix:kgap}
\begin{table}
\caption{Abbreviations (KGAP)}
\begin{tabular}{p{0.2\linewidth}p{0.8\linewidth}}
\hline
\makecell[r]{\textbf{ATC}} & Anatomical Therapeutic Chemical Classification (WHO)\\
\makecell[r]{\textbf{AUROC}} & Area under receiver operator characteristic (ROC) curve\\
\makecell[r]{\textbf{IDG}} & Illuminating the druggable genome (NIH Common Fund)\\
\makecell[r]{\textbf{KG}} & Knowledge graph, also known as a knowledge network\\
\makecell[r]{\textbf{KGAP}} & Knowledge graph analytics platform\\
\makecell[r]{\textbf{LINCS1000}} & Landmark genes (approximately 1000) from LINCS, chosen for maximal inference of full genomic expression.\\
\makecell[r]{\textbf{LINCS}} & Library of integrated network-based cellular signatures (NIH Common Fund)\\
\makecell[r]{\textbf{MoA}} & Mechanism of action, describing biomolecular details of drug effect\\
\makecell[r]{\textbf{PD}} & Parkinson's disease\\
\makecell[r]{\textbf{SYNGR3}} & Synaptogyrin-3, a human gene, subject of case study in this paper\\
\makecell[r]{\textbf{TCRD}} & Target central resource database (IDG)\\
\makecell[r]{\textbf{TDL}} & Target development level (IDG)\\
\makecell[r]{\textbf{TIN-X}} & Target Importance and Novelty Explorer (IDG)\\
\hline
\end{tabular}
\end{table}

\section{TIGA}
\label{appendix:tiga}
\begin{table}
\caption{Abbreviations and Definitions (TIGA)}
\begin{tabular}{p{0.2\linewidth}p{0.8\linewidth}}
\hline
\multicolumn{2}{p{1.0\linewidth}}{\textit{Common terms used in GWAS and related fields can vary in their definitions and connotations depending on context. Therefore for clarity and rigor the following definitions are provided, which we consider consistent with best practices in the GWAS and drug discovery communities.}}\\
\hline
\makecell[r]{\textbf{genotype}} & An organism has one genotype, comprised of a germ line genome and multiple somatic genomes. Statistical models may assume a population distribution hence a population genotype.\\
\makecell[r]{\textbf{phenotype}} & An organism has one phenotype, comprised of (potentially) all non-genomic observable characteristics, a.k.a. phenotypic traits.\\
\makecell[r]{\textbf{gene}} & Genomic unit responsible for an expression product. Protein coding genes are a subset of this definition.\\
\makecell[r]{\textbf{trait}} & Single non-genomic, observable characteristic.\\
\makecell[r]{\textbf{drug target}} & Biomolecular entity involved in the mechanism of action of a drug. The IDG project is protein-centric; hence drug targets are all proteins.\\
\hline
\end{tabular}
\end{table}

\chapter{Glossary of data science terms}

\label{appendix:glossary_datascience}

\input{appendices/glossary_interpretation.tex}

\input{appendices/glossary_semantics.tex}

\input{appendices/glossary_wrangling.tex}

\input{appendices/glossary_semantics.tex}

\input{appendices/glossary_stats.tex}

\input{appendices/glossary_visualization.tex}

% \input{appendices/glossary_ml.tex}

% \input{appendices/glossary_packages.tex}

\input{appendices/glossary_misc.tex}

\end{appendices}
