\chapter{TIGA (Target illumination GWAS analytics)}

\section{Abstract}

Genome wide association studies (GWAS) can reveal important genotype–phenotype associations, however, data quality and interpretability issues must be addressed. For drug discovery scientists seeking to prioritize targets based on the available evidence, these issues go beyond the single study. Here, we describe rational ranking, filtering and interpretation of inferred gene–trait associations and data aggregation across studies by leveraging existing curation and harmonization efforts. Each gene–trait association is evaluated for confidence, with scores derived solely from aggregated statistics, linking a protein-coding gene and phenotype. We propose a method for assessing confidence in gene–trait associations from evidence aggregated across studies, including a bibliometric assessment of scientific consensus based on the iCite Relative Citation Ratio, and meanRank scores, to aggregate multivariate evidence. This method, intended for drug target hypothesis generation, scoring and ranking, has been implemented as an analytical pipeline, available as open source, with public datasets of results, and a web application designed for usability by drug discovery scientists, at https://unmtid-shinyapps.net/tiga/[41].

\section{Keywords}

GWAS, data science, drug discovery, drug target, druggable genome

\section{Introduction}

Over the two decades since the first draft human genome was published, dramatic progress has been achieved in foundational biology with translational benefits to medicine and human health. Genome wide association studies (GWAS) contribute to this progress by inferring associations between genomic variations and phenotypic traits [42,43]. These associations are correlations which may or may not be causal. While GWAS can reveal important genotype–phenotype associations, data quality and interpretability must be addressed [44–47].  For drug discovery scientists seeking to prioritize targets based on evidence from multiple studies, quality and interpretability issues are broader than for GWAS specialists. For this use case, GWAS are one of several evidence sources to be explored and considered, and interpretability must be in terms of genes corresponding to plausible targets, and traits corresponding to diseases of interest.

Single nucleotide variants (SNV) are the fundamental unit of genomic variation, and the term single nucleotide polymorphism (SNP) refers to SNVs identified as common sites of variation relative to a reference genome, and measured by microarray or sequencing technologies. The NHGRI-EBI GWAS Catalog [48] -- hereafter "Catalog" -- curates associations between SNPs and traits from GWAS publications, shares metadata and summary data, standardizes heterogeneous submissions, maps formats and harmonizes content, mitigating widespread data and meta-data issues according to FAIR (Findable, Accessible, Interoperable and Reusable) principles [49]. These challenges are exacerbated by rapid advances in experimental and computational methodology. As de facto GWAS registrar, the Catalog interacts directly with investigators and accepts submissions of summary statistic data in advance of publication. Proposing and maintaining metadata standards the Catalog advocates and advances FAIRness in GWAS, for the benefit of the community. The Catalog addresses many difficulties due to content and format heterogeneity, but there are continuing difficulties and limitations both from lack of reporting standards and the variability of experimental methodology and diagnostic criteria.

Other GWAS data collections include the Genome-Wide Repository of Associations between SNPs and Phenotypes, GRASP [50] and The Framingham Heart Study, which employs non-standard phenotypes and some content from the Catalog (not updated since 2015). GWASdb [51] integrates over 40 data sources in addition to the Catalog, includes less significant variants to address a variety of use cases, and has been maintained continually since 2011. GWAS Central, continually updated through 2019, includes less significant associations and provides tools for a variety of exploration modes based on Catalog data, but is not freely available for download. PheGenI [52] integrates Catalog data with other NCBI datasets and tools. Others integrate GWAS with additional data (e.g. pathways, expression, linkage disequilibrium) to associate traits or diseases with genes [20,53–56]. Each of these resources offers unique value and features. For this use case, the Catalog is the logical choice, given its applicability and commitment to expert curation, data standards, support and maintenance.

Here we describe TIGA (Target Illumination GWAS Analytics), an application for illuminating understudied drug targets. TIGA enables ranking, filtering and interpretation of inferred gene-trait associations aggregated across studies from the Catalog. Each inferred gene-to-trait association is evaluated for confidence, with scores derived solely from evidence aggregated across studies, linking a phenotypic trait and protein-coding gene, mapped from single nucleotide polymorphism (SNP) variation. TIGA uses the Relative Citation Ratio, RCR [57], a bibliometric statistic from iCite [58]. TIGA does not index the full corpus of GWAS associations, but focuses on the strongest associations at the protein-coding gene level instead, filtered by disease areas that are relevant to drug discovery. For instance, GWAS for highly polygenic traits are considered less likely to illuminate druggable genes. Here, we describe the web application and its interpretability for non-GWAS specialists. We discuss TIGA as an application of data science for scientific consensus and interpretability, including statistical and semantical challenges. Code and data are available under BSD-2-Clause license from https://github.com/unmtransinfo/tiga-gwas-explorer.

\section{Methods}

\subsection{NHGRI-EBI GWAS Catalog preprocessing}

The 2021-02-12 release of the Catalog references 11671 studies and 4865 PubMed IDs. The curated associations include 8235 studies and 2706 EFO-mapped traits. After filtering studies to require (1) mapped trait, (2) p-value below 5e-8, (3) reported effect size (odds-ratio or beta), and (4) mapped protein-coding gene, we obtain 4118 studies, 1521 traits, and 10264 genes. For consistency between studies, only genes mapped by the Ensembl pipeline for genomics annotations were considered (not author-reported). Figures \ref{fig:TIGA_Study_coun} and \ref{fig:TIGA_Samp_sizes} illustrate the growth of GWAS research as measured by counts of studies and subjects.  

\begin{figure}
	\includegraphics[width=\textwidth]{figures/tiga/FIG01_Study_counts.png}
	\caption{GWAS Catalog study counts by year and vendor, indicating growth and platform trends.}
	\label{fig:TIGA_Study_coun}
\end{figure}

\begin{figure}
	\includegraphics[width=\textwidth]{figures/tiga/FIG02_Sample_sizes.png}
	\caption{GWAS Catalog sample size distributions by year, on log scale, indicating variance in statistical power.}
	\label{fig:TIGA_Samp_sizes}
\end{figure}

\subsection{RCRAS = Relative Citation Ratio (RCR) Aggregated Score}

The purpose of TIGA is to evaluate the evidence for a gene-trait association, by aggregating multiple studies and their corresponding publications.  The iCite RCR [57] is a bibliometric statistic designed to evaluate the impact of an individual publication (in contrast to the journal impact factor).  By field- and time-normalizing per-publication citation counts, the RCR measures evolving impact, in effect a proxy for scientific consensus.  Hence by aggregating RCRs we seek a corresponding measure of scientific consensus for associations.

\begin{equation} RCRAS_{gt} = \sum_{study} \left(\frac{1}{gc} \sum_{pub} \frac{log_{2}(RCR + 1)}{sc}\right)
\end{equation}

Where \\
\begin{center}
\begin{tabular}{ r c l }
    $study$ & $=$ & GWAS (study\_accession) \\
	$gc$ & $=$ & gene count (in study)	\\
	$pub$ & $=$ & publication (PubMed ID)	\\
	$sc$ & $=$ & study count (in publication)	\\
\end{tabular}
\end{center}

The $log_2()$ function is used with the assertion that differences of evidence depend on relative, rather than absolute differences in RCR. Division by sc effects a partial count for publications associated with multiple studies.  Since $RCR \geq 0$, $log_2(RCR + 1) \geq 0$ and intuitively, when $RCR = 1$ and $sc = 1$, $log_2(RCR + 1) = 1$. Similarly division by gc reflects a partial count since studies may implicate multiple genes. This approach is informed by bibliometric methodology, including fractional counting, as described elsewhere [11]. For recent publications lacking RCR, we used the global median as an estimated prior. Computed thus, RCRAS extends RCR with similar logic, providing a rational bibliometric measure of evidence for scoring and ranking gene-trait associations. 

\subsection{Association weighting by SNP–gene distance}

Mapping  genomic  variation  of  single  nucleotides  (SNPs)  to  genes  is  a  challenging  area  of active  research [59–61].  This project  does not contribute to mapping methodology.  Rather, TIGA employs mappings provided by the Catalog between GWAS SNPs and genes,  generated by an Ensembl pipeline, which "adds additional SNP specific information associated with the rsID extracted... This information is retrieved using the Ensembl API and the source of the data is both Ensembl and NCBI."[62] It is important to note that this method is unbiased and derived from experimental data and the current human reference genome . TIGA aggregates SNP-trait associations, assessing evidence for gene-trait associations, based on these understandings:

\begin{itemize}
\item SNPs within a gene are more strongly associated than SNPs upstream or downstream.
\item Strength of association decreases with distance, or more rigorously stated, the probability of linkage disequilibrium (LD) between a SNP and protein coding gene decreases with genomic physical distance. Accordingly, we employ an inverse exponential scoring function, consistent with LD measure ($\Delta$) and coefficient of decay ($B$) by Wang and coworkers [63]. 
\end{itemize}

This function, used to weight N\_snp to compute a distance-weighted SNP count N\_snpw, is plotted together with the observed frequencies of mapped gene distances in supplementary Fig. 1, to illustrate how the extant evidence is weighted. 

\begin{equation}
    N\_snpw = \sum_{i}^{N\_snp}  2^{-d_{i}/k}
\end{equation}

\begin{center}
    where $ d = $ \emph{distance in base pairs} \\
    and $ k = $ \emph{"half-life distance"} (50k)
\end{center}

\subsection{Multivariate ranking}

Multivariate ranking is a well studied problem which needs to be addressed for ranking GWAS associations. We evaluated two approaches, namely non-parametric $\mu$ scores (Wittkowski, 2008) and meanRank, and chose the latter based on benchmark test performance. meanRank aggregates ranks instead of variables directly, avoiding the need for ad hoc parameters. Variable-ties imply rank-ties, with missing data ranked last. We normalize scoring to (0,100] defining meanRankScore as follows.

Variables of merit used for scoring and ranking gene-trait associations:

\begin{itemize}
\item N\_snpw: N\_snp weighted by distance inverse exponential described above.
\item pVal\_mLog: $max(-Log(pValue))$ supporting gene-trait association.
\item RCRAS: Relative Citation Ratio (RCR) Aggregated Score (iCite-RCR-based), described above.
\end{itemize}

Variables of merit and interest not currently used for ranking:

\begin{itemize}
\item OR: median(odds ratio, inverted if <1) supporting gene-trait association.
\item N\_beta: simple count of beta values with 95% confidence intervals supporting gene-trait association.
\item N\_snp: SNPs involved with gene-trait association.
\item N\_study: studies supporting gene-trait association.
\item study\_N: mean(SAMPLE\_SIZE) supporting gene-trait association.
\item geneNtrait: total traits associated with the gene.
\item traitNgene: total genes associated with the trait.
\end{itemize}

N\_snp, N\_study, geneNtrait and traitNgene are counts of the corresponding unique entities. From the variables selected via benchmark testing the meanRankScore is computed thus:

\begin{equation}
    meanRankScore = 100 - Percentile(meanRank) = 100 - Percentile(\frac{1}{N} \sum_{i}^{N} rank_{i})
\end{equation}

\begin{center}
    where $ rank_{i} = $ \emph{rank of $ i $th variable} \\
    and $ N = $ \emph{number of variables considered}
\end{center}

$\mu$ scores were implemented via the muStat [64] R package. Vectors of ordinal variables represent each case, and non-dominated solutions are cases which are not inferior to any other case at any variable. (For TIGA, cases are genes or traits, corresponding with trait-queries or gene-queries, respectively, and their variables of merit described above.) The set of all non-dominated solutions defines a Pareto-boundary. A $\mu$ score is defined simply as the number of lower cases minus the number of higher, but the ranking is the useful result. The ranking rule between case $k$ and case $k'$ may be formalized thus:

\begin{equation}
    \{ x_k < x_{k'} \} \Leftrightarrow \{ \forall_{l=1...L} x_{kl} \leq x_{k'l} \wedge \exists_{l=1...L} x_{kl} < x_{k'l} \}
\end{equation}


Simply put, case $k’$ is higher than case $k$ if it is higher in some variable(s) and lower in none.

\subsection{Benchmark against gold standard}

Lacking a suitable gold standard set of gene–trait associations in general, we instead relied on established gene–disease associations from the Genetics Home Reference, GHR [65] and UniProtKB [66] databases. This gold stand set was built following a previously described approach [67]. It consists of 5,366 manually curated associations (positive examples) between 3,495 genes and 709 diseases. All other (2,472,589) possible pairings of these genes and diseases were considered negative examples.

To assess the quality of the TIGA gene–trait associations, we mapped the Ensembl gene IDs to STRING v11 identifiers using the STRING alias file [23] and the EFO terms to Disease Ontology identifiers [68] based on ontology cross-references and the EMBL-EBI Ontology Xref Service. We then benchmark any individual-variable or multivariate ranking of the associations by constructing the receiver operating characteristic (ROC) curve by counting the agreement with the gold standard.

\section{Results}

\subsection{The TIGA web application}

\begin{figure}
	\includegraphics[width=\textwidth]{figures/tiga/FIG03_TIGA_EFO_0004612_plot.png}
	\caption{TIGA web application at https://unmtid-shinyapps.net/tiga/, displaying a plot of 644 genes currently associated with trait "high density lipoprotein cholesterol measurement" (EFO\_0004612).}
	\label{fig:TIGA_app}
\end{figure}

TIGA facilitates drug target illumination by currently scoring and ranking associations between protein-coding genes and GWAS traits. While not capturing the entire Catalog, the TIGA app can aggregate and filter GWAS findings for actionable intelligence, e.g., to enrich target prioritization via interactive plots and hitlists (figure \ref{fig:TIGA_app}), allowing users to identify the strongest associations supported by evidence.

Hits are ranked by meanRankScore described in Methods. Scatterplot axes are Effect (OR or N\_beta) vs. Evidence as measured by meanRankScore. Plot markers may be sized by pValue or RCRAS. This app accepts "trait" and "gene" query parameters via URL, e.g. ?trait=EFO\_0004541, ?gene=ENSG00000075073, ?trait=EFO\_0004541\&gene=ENSG00000075073.  Gene markers are colored by Target Development Level (TDL)[3]. TDL is a knowledge-based classification that bins human proteins into four categories: Tclin, mechanism-of-action designated targets via which approved drugs act [69–71]; Tchem are proteins known to bind small molecules with high potency; Tbio includes proteins that have Gene Ontology [72] “leaf” (lowest level) experimental terms; or meet two of these conditions:  A fractional publication count [73] above 5, three or more Gene “Reference Into Function” annotations [74], or 50 or more commercial antibodies in Antibodypedia [75]; Tdark are manually curated UniProtKB proteins that fail to place in any of the previous categories.

\begin{figure}
	\includegraphics[width=\textwidth]{figures/tiga/FIG04_ROCs.png}
	\caption{Performance evaluation. The performance of TIGA on the gold standard of gene-disease associations. A) Results for the top-3 individual variables of merit. B) Results for the multivariate ranking by meanRankScore and $\mu$ score.}
	\label{fig:TIGA_rocs}
\end{figure}

\subsection{Using TIGA for drug target illumination}

The main motivation of developing TIGA is to capture GWAS data when illuminating drug targets. Table 1 shows how many targets from each protein family and IDG Target Development Level (TDL) are covered with associated traits in TIGA, with families as defined by Drug Target Ontology[76] (DTO) Level 2. IDG TDL is a knowledge based classification: Tclin = high-confidence drug targets; Tchem = small-molecule modulator exists; Tbio = biological function elucidated; Tdark = minimal knowledge [3]. Coverage for the understudied 2,469 Tdark proteins is of particular interest. However, the data for other TDLs can also provide unique and complementary evidence, especially in case of Tbio proteins that are biologically characterized but have not before been clinically validated.


\begin{table}
\caption{TIGA mapped target (protein) counts by IDG Target Development Level (TDL) and Drug Target Ontology (DTO) level 2 gene family.}
\begin{center}
\begin{tabular}{ |l|c|c|c|c|c| } 
\hline

\hline
 
\hline
\end{tabular}
\end{center}
\label{table:idg_counts}
\end{table}

Family \ TDL
Tclin
Tchem
Tbio
Tdark
Total
G-protein coupled receptor
73 / 101
78 / 143
73 / 129
110 / 407
334 / 780
Ion channel
97 / 127
59 / 89
72 / 116
12 / 20
240 / 352
Kinase
57 / 66
278 / 360
97 / 133
12 / 20
444 / 579
Calcium-binding protein
3 / 5
1 / 3
58 / 93
8 / 11
70 / 112
Cell-cell junction
0 / 0
0 / 0
22 / 49
8 / 12
30 / 61
Cell adhesion
0 / 1
0 / 2
23 / 52
6 / 15
29 / 70
Cellular structure
4 / 10
5 / 11
244 / 323
44 / 86
297 / 430
Chaperone
0 / 1
8 / 9
27 / 46
6 / 8
41 / 64
Enzyme modulator
4 / 5
25 / 44
376 / 532
50 / 101
455 / 682
Enzyme
69 / 104
277 / 387
1022 / 1553
177 / 332
1545 / 2376
Epigenetic regulator
9 / 13
41 / 55
16 / 22
0 / 1
66 / 91
Extracellular structure
0 / 1
0 / 1
50 / 57
8 / 9
58 / 68
Immune response
0 / 1
0 / 2
13 / 41
4 / 6
17 / 50
Nuclear receptor
16 / 18
16 / 19
8 / 11
0 / 0
40 / 48
Nucleic acid binding
0 / 1
13 / 19
354 / 603
67 / 131
434 / 754
Transcription factor
1 / 2
12 / 16
385 / 557
73 / 163
471 / 738
Transporter
31 / 37
63 / 82
405 / 605
105 / 160
604 / 884
Receptor
20 / 24
6 / 12
157 / 225
27 / 55
210 / 316
Signaling
13 / 24
24 / 32
245 / 338
17 / 34
299 / 428
Storage
0 / 1
0 / 1
2 / 7
1 / 2
3 / 11
Surfactant
0 / 0
0 / 0
3 / 5
0 / 0
3 / 5
Other
95 / 134
233 / 337
3973 / 6131
1734 / 3416
6035 / 10018
Total
492 / 676
1139 / 1624
7625 / 11628
2469 / 4989
11725 / 18917

